File: src/main/java/work/stdpi/pge/editor/modmenu/PgeModMenuIntegration.java
package work.stdpi.pge.editor.modmenu;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;

public class PgeModMenuIntegration implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return PgeSettingsScreen::new;
    }
}

File: src/main/kotlin/work/stdpi/pge/editor/modmenu/PgeModMenuIntegration.kt
package work.stdpi.pge.editor.modmenu

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import net.minecraft.client.gui.screen.Screen

class PgeModMenuIntegration : ModMenuApi {
  override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
    return ConfigScreenFactory { parent: Screen? -> PgeSettingsScreen(parent) }
  }
}
